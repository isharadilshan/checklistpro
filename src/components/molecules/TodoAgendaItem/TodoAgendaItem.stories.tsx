import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {ComponentMeta, ComponentStory} from '@storybook/react';

import TodoAgendaItem from './';

export default {
  title: 'components/molecules/TodoAgendaItem',
  component: TodoAgendaItem,
} as ComponentMeta<typeof TodoAgendaItem>;

export const Basic: ComponentStory<typeof TodoAgendaItem> = () => {
  const args = {
    height: 150,
    data: 'Agenda Title@Agenda Description@PERSONAL@DONE',
  };
  return (
    <NativeBaseProvider>
      <TodoAgendaItem {...args} />
    </NativeBaseProvider>
  );
};
