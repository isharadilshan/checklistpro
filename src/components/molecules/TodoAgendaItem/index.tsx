import React from 'react';
import {Badge, View, Text} from 'native-base';
import createStyle from './styles';

type TodoAgendaItemProps = {
  data: string;
  height: number;
};

const TodoAgendaItem: React.FC<TodoAgendaItemProps> = ({data, height}) => {
  const styles = createStyle();

  const renderBadge = (colorScheme: string, status: string) => {
    return (
      <Badge mt={4} width={100} variant={'subtle'} colorScheme={colorScheme}>
        {status}
      </Badge>
    );
  };

  const renderStatusBadge = (todoStatus) => {
    switch (todoStatus) {
      case 'TODO':
        return renderBadge('coolGray', todoStatus);
      case 'INPROGRESS':
        return renderBadge('warning', todoStatus);
      case 'HOLD':
        return renderBadge('error', todoStatus);
      case 'DONE':
        return renderBadge('info', todoStatus);
      default:
        return null;
    }
  };

  return (
    <View style={styles.agendaItemWrapper}>
      <Text color={'blueGray.400'} fontSize="xl">
        {data.split('@')[0]}
      </Text>
      <Text color={'blueGray.400'}>{data.split('@')[1]}</Text>
      <View style={styles.tagWrapper}>
        <Badge mt={4} width={100} variant={'solid'} colorScheme="success">
          {data.split('@')[2]}
        </Badge>
        {renderStatusBadge(data.split('@')[3])}
      </View>
    </View>
  );
};

export default TodoAgendaItem;
