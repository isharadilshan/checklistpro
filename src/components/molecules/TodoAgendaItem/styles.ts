import {StyleSheet, ViewStyle} from 'react-native';

interface Style {
  agendaItemWrapper: ViewStyle;
}

export default () => {
  return StyleSheet.create<Style>({
    agendaItemWrapper: {
      flex: 1,
      borderRadius: 4,
      paddingVertical: 20,
      paddingHorizontal: 10,
      marginVertical: 20,
      marginHorizontal: 10,
      backgroundColor: '#141E30',
    },
  });
};
