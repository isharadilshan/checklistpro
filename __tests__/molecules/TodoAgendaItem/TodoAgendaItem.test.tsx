import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TodoAgendaItem from '../../../src/components/molecules/TodoAgendaItem';

configure({adapter: new Adapter()});

const props = {
  data: 'Agenda Title@Agenda Description@PERSONAL@DONE',
};

describe('when rendering', () => {
  const wrapper = shallow(<TodoAgendaItem {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
