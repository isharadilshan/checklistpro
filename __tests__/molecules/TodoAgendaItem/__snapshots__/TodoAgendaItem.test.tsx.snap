// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<ForwardRef
  style={
    {
      "backgroundColor": "#141E30",
      "borderRadius": 4,
      "flex": 1,
      "marginHorizontal": 10,
      "marginVertical": 20,
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <Memo(ForwardRef(Text))
    color="blueGray.400"
    fontSize="xl"
  >
    Agenda Title
  </Memo(ForwardRef(Text))>
  <Memo(ForwardRef(Text))
    color="blueGray.400"
  >
    Agenda Description
  </Memo(ForwardRef(Text))>
  <ForwardRef
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <Memo(ForwardRef(Badge))
      colorScheme="success"
      mt={4}
      variant="solid"
      width={100}
    >
      PERSONAL
    </Memo(ForwardRef(Badge))>
    <Memo(ForwardRef(Badge))
      colorScheme="info"
      mt={4}
      variant="subtle"
      width={100}
    >
      DONE
    </Memo(ForwardRef(Badge))>
  </ForwardRef>
</ForwardRef>
`;
